### ISSUE TITLE
Passing Jackett container network through transmission-openvpn container

### ISSUE THREAD

**Issue by adamburgoyne** [haugene/docker-transmission-openvpn#991]
Hey, thanks for the work! I've got it all setup and running now with transmission going through my PIA VPN. I'm looking to now pass the Jackett containers traffic through the the same VPN but since the VPN is embedded within the transmission container I'm not sure how to do it via docker-compose? I did try splitting out transmission from the vpn but had issues with running openvpn on Raspbian.

I did manage to route by adding "network-mode: 'services:transmission'" to jackett but that meant I wasn't able to access the jackett UI anymore. Any ideas? Thanks!

Here is the main parts of my docker-compose.yml:

```
version: "2"

services:
  transmission:
    image: haugene/transmission-openvpn:latest-armhf
    cap_add:
      - NET_ADMIN
    devices:
      - "/dev/net/tun"
    container_name: transmission-openvpn
    restart: always
    ports:
      - "9091:9091"
      - "8888:8888"
    dns:
      - 8.8.8.8
      - 8.8.4.4
    networks:
      - plexnet
    environment:
      - PGID=1001
      - PUID=1001
      - OPENVPN_PROVIDER=${VPN_PROVIDER}
      - OPENVPN_CONFIG=${VPN_CONFIG}
      - OPENVPN_USERNAME=${VPN_USERNAME}
      - OPENVPN_PASSWORD=${VPN_PASSWORD}
      - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
      - LOCAL_NETWORK=192.168.0.0/16
    volumes:
      - ${MOUNT_POINT}/transmission:/data
      - ${MOUNT_POINT}/transmission/config:/config
      - ${MOUNT_POINT}/transmission/watch:/watch
      - /etc/localtime:/etc/localtime:ro
  jackett:
    image: linuxserver/jackett
    container_name: jackett
    restart: always
    networks:
      - plexnet
    environment:
      - PGID=1001
      - PUID=1001
      - TZ=Europe/London
    volumes:
      - ${MOUNT_POINT}/jackett/config:/config
      - ${MOUNT_POINT}/jackett/downloads:/downloads
    ports:
      - "9117:9117"
networks:
  plexnet:
    driver: bridge
```

**Comment by adamburgoyne** [comment]
I have tried adding the Jackett port `- "9117:9117"` to the list of transmission IPs and add `network_mode: "service:transmission"` to the Jackett container. I can see Jackett starting okay in the logs but I'm unable to access it via the UI under port 9117

**Comment by adamburgoyne** [comment]
Ah found it, Jackett was just taking an age to start up compared to everything else and I was being inpatient. For anyone else looking here's my docker-compose.

```
version: "2"

services:
  transmission:
    image: haugene/transmission-openvpn:latest-armhf
    cap_add:
      - NET_ADMIN
    devices:
      - "/dev/net/tun"
    container_name: transmission-openvpn
    restart: always
    ports:
      - "9091:9091"
      - "8888:8888"
      - "9117:9117"
    dns:
      - 8.8.8.8
      - 8.8.4.4
    networks:
      - plexnet
    environment:
      - PGID=1001
      - PUID=1001
      - OPENVPN_PROVIDER=${VPN_PROVIDER}
      - OPENVPN_CONFIG=${VPN_CONFIG}
      - OPENVPN_USERNAME=${VPN_USERNAME}
      - OPENVPN_PASSWORD=${VPN_PASSWORD}
      - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
      - LOCAL_NETWORK=192.168.0.0/16
    volumes:
      - ${MOUNT_POINT}/transmission:/data
      - ${MOUNT_POINT}/transmission/config:/config
      - ${MOUNT_POINT}/transmission/watch:/watch
      - /etc/localtime:/etc/localtime:ro
  jackett:
    image: linuxserver/jackett
    container_name: jackett
    restart: always
    network_mode: "service:transmission"
    environment:
      - PGID=1001
      - PUID=1001
      - TZ=Europe/London
    volumes:
      - ${MOUNT_POINT}/jackett/config:/config
      - ${MOUNT_POINT}/jackett/downloads:/downloads
networks:
  plexnet:
    driver: bridge
```

**Comment by jt196** [comment]
@adamburgoyne I have this running, however, a couple of quick questions about this set up:

1. Is there any way to port forward the other containers hiding behind the TVPN setup? Is it a matter of setting the same port as Transmission has. 
2. If the port changes every time Transmission connects, is there any way to automate the port for the other containers?
3. I've found some issues with start up sequences with Watchtower updating the Transmission container. Is there any recommended setting you've put for this? I've used depends on but it still breaks the other containers until a restart.
